corgi/core/models.py

```diff
@@ -46,6 +51,7 @@ class MPTTMeta:
     #
     # - Each method assumes that an object model will always link to a single ProductNode (thus
     #   the use of `pnodes.first()`).
+    #   TODO: Need to revisit, we have ProductVariants which are linked to two separate streams
```

Comment:
in this case each child ProductNode from the two streams would just point to the same ProductVariant entity

Reply:
Yes, that's what I was proposing here. I thought we could add a for loop which replaces the .first(), in order to process all the related ProductNodes instead of only the first one.

If we do that, though, two different trees will point to the same ProductVariant, which might cause issues elsewhere if we don't expect a ProductVariant to be in two different trees.

I wasn't proposing any model changes, just noting that our tree will no longer look like a tree and trying to find e.g. all linked ProductStreams may return odd results.